src/Core/Silk.NET.BuildTools/Overloading/Simple/RefOverloader.cs

```diff
@@ -10,10 +11,25 @@ namespace Silk.NET.BuildTools.Overloading
 {
     public class RefOverloader : ISimpleParameterOverloader, IPostProcessingSimpleOverloader
     {
-        public bool TryGetParameterVariant(Parameter parameter, out Parameter varied, Profile _)
+        public bool TryGetParameterVariant(Parameter parameter, out Parameter varied, Profile profile)
         {
             if (parameter.Type.IsPointer)
             {
+                //if its an opaque pointer, dont do ref overloads
+                if(parameter.Type.IndirectionLevels == 1)
+                {
+                    var baseType = new TypeSignatureBuilder(parameter.Type).WithIndirectionLevel(0).Build();
+
+                    foreach (var project in profile.Projects)
+                    {
+                        if(project.Value.Structs.FirstOrDefault(x => x.Name == baseType.Name)?.IsOpaque ?? false)
```

Comment:
This is equivalent to just doing parameter.Type.Name, why do we need the TypeSignatureBuilder?

Reply:
didnt know that, fixed